src/main/java/software/amazon/event/kafkaconnector/mapping/JsonPathDetailTypeMapper.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package software.amazon.event.kafkaconnector.mapping;
+
+import static com.jayway.jsonpath.Configuration.defaultConfiguration;
+import static com.jayway.jsonpath.Option.SUPPRESS_EXCEPTIONS;
+import static java.lang.String.format;
+import static java.util.Collections.singletonMap;
+import static org.apache.kafka.connect.json.JsonConverterConfig.SCHEMAS_ENABLE_CONFIG;
+import static software.amazon.event.kafkaconnector.EventBridgeSinkConfig.AWS_DETAIL_TYPES_MAPPER_JSON_PATH_MAPPER_FIELDREF;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.jayway.jsonpath.Configuration;
+import com.jayway.jsonpath.JsonPath;
+import org.apache.kafka.connect.json.JsonConverter;
+import org.apache.kafka.connect.sink.SinkRecord;
+import org.slf4j.Logger;
+import software.amazon.event.kafkaconnector.EventBridgeSinkConfig;
+import software.amazon.event.kafkaconnector.logging.ContextAwareLoggerFactory;
+
+public class JsonPathDetailTypeMapper implements DetailTypeMapper {
+  private static final Logger log =
+      ContextAwareLoggerFactory.getLogger(JsonPathDetailTypeMapper.class);
+
+  private static final Configuration jsonPathConfiguration =
+      defaultConfiguration()
+          .addOptions(
+              SUPPRESS_EXCEPTIONS // suppress exception otherwise
+              // com.jayway.jsonpath.ReadContext#read throws an exception if JSON path could
+              // not
+              // be found
+              );
+
+  private String jsonPath;
+  private final JsonConverter jsonConverter;
+  private final ObjectMapper objectMapper;
+
+  public JsonPathDetailTypeMapper() {
+    this.jsonConverter = new JsonConverter();
+    this.objectMapper = new ObjectMapper();
+    this.jsonConverter.configure(singletonMap(SCHEMAS_ENABLE_CONFIG, "false"), false);
+  }
+
+  @Override
+  public String getDetailType(SinkRecord record) {
+    if (record == null) {
+      throw new IllegalArgumentException("SinkRecord is null. Unable to extract detail type.");
+    }
+
+    if (record.topic() == null || record.topic().trim().isEmpty()) {
+      throw new IllegalArgumentException(
+          "SinkRecord topic is null or empty but is required for fallback logic.");
+    }
+
+    var topic = record.topic();
+
+    try {
+      var jsonBytes = jsonConverter.fromConnectData(topic, record.valueSchema(), record.value());
+      if (jsonBytes == null) {
+        // TODO: discuss throw or error
```

Comment:
@agebhar1 any thoughts on the TODOs left here?

Reply:
@maschnetwork same question for you.

Reply:
What is the exact question here? Can you provide an example?

Reply:
There's currently two cases (TODOs) in the code where I wasn't sure whether I should throw or log an error and use the fallback (topic name). My proposal is to log an error but wanted to seek feedback here.

I think `if (jsonBytes == null)` case can be ignored in the discussion because it's just defensive code which should not happen in production.

the other case is when catching `IllegalArgumentException` (not sure which cases would trigger it, this was recommended by @agebhar1 )

```java
catch (IllegalArgumentException e) {
      // TODO: discuss throw or error
      log.error(
          "Could not extract JSON value for JSON path '{}' and record '{}', using topic '{}' as fallback",
          jsonPath,
          record,
          topic);
      return topic;
```

Reply:
See comment below on the catch clause - we can potentially remove this and should also replace the catch all and rethrow at the bottom with log.error and returning the fallback.

In general I would propose to not throw in this method at all - either the JsonPath works or we have a fallback. Maybe even a `.warn` instead of `.error` for the fallback cases that we are covering explicitly.

Maybe in the rare circumstance that a topic is null and the fallback does not work we can keep the throw.